### PR TITLE
Early talker

### PR DESCRIPTION
**Pull request by msimerson**
contains @smfreegard 's #1057 squashed:

* set early_talker to delay at connect_init and data …
* Remove early_talker delay in core

and a few of my changes:

* added reject option
* save pass/fail results
* move config to early_talker.ini
    * with fallback to early_talker.pause
* added reject option (default: true)
* only calculate elapsed when hook=connect_init
* when pausing, subtract elapsed from pause
* make sure connection exists after delay (else a CRIT happens)
* brought docs the rest of the way up-to-date
* renamed the test file to match the plugin name

Closes #1057

### CODE REVIEW

File: config/early_talker.ini [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+; delay in seconds
+pause=5
```
Comment (smfreegard) [inline config/early_talker.ini]: I don't think this should be in seconds.   My default delay here is 500ms...

**Comment by baudehlo** [comment]
There's a comment in #1075 about a change to milliseconds. Is this a block on merging this?

**Comment by baudehlo** [comment]
Sorry, #1065 I mean.

**Comment by msimerson** [comment]
@smfreegard thinks so. I think seconds is far more user-friendly, but I understand Steve's desire to use portions of seconds. Even after this change, he can continue to use milliseconds in `config/early_talker.pause`.

**Comment by baudehlo** [comment]
Not reading the code right now so two questions:

1) is that a breaking change?

2) does it support decimal seconds?

If it supports decimal seconds and doesn't break things then it seems fine.

**Comment by msimerson** [comment]
Non-breaking.

Yes, it certainly supports decimal seconds.

````
 $ node 
> 0.5 * 1000
500
> '0.5' * 1000
500
````

**Comment by msimerson** [comment]
Having thought more about this, I feel that values in config files really **ought**  to be seconds. 

Milliseconds are still the default in `early_talker.pause` and are also available as decimal values in the `.ini` file.

**Pull request merged**